pkg/webhook/injection.go

```diff
@@ -71,6 +71,37 @@ func injectSidecarContainer(pod *corev1.Pod, config *Config, supportsNativeSidec
 	}
 }
 
+func (si *SidecarInjector) injectMetadataPrefetchSidecarContainer(pod *corev1.Pod, config *Config, supportsNativeSidecar bool) {
+	var containerSpec corev1.Container
+	var index int
+
+	if supportsNativeSidecar {
```

Comment:
we might be using criteria when defining if the workload should be using supportsNativeSidecar. We need to verify the criteria is the same throughout.

Reply:
this criteria is the same for now, but will change in #351